File: agent/commands-generator/src/main/java/org/matilda/commands/python/writer/PythonVariable.java
package org.matilda.commands.python.writer;

public record PythonVariable(String name, String typeHint) {
    public PythonVariable(String name) {
        this(name, null);
    }

    public String getDeclaration() {
        if (typeHint == null) {
            return name;
        }
        return name + ": " + typeHint;
    }
}

File: agent/commands-generator/src/main/java/org/matilda/commands/python/writer/PythonVariable.kt
package org.matilda.commands.python.writer

data class PythonVariable(val name: String, val typeHint: String?) {
    constructor(name: String) : this(name, null)

    val declaration: String
        get() = if (typeHint == null) {
            name
        } else "$name: $typeHint"
}
